request_library.md

```diff
@@ -0,0 +1,223 @@
+
+```
+${pin}= Get Request gpio_server /api/v1/pin/18 ${headers}
+```
+
+As I said, `alias` is very important. Unfortunatelly for observing results
+Robot Framework has special keyword. Its name is `Log`.
+
+Whole test look like this:
+
+```
+*** Settings ***
+Library RequestsLibrary
+Library Collections
+
+
+*** Variables ***
+${URL} http://192.168.0.46:5000
+
+
+*** Test Cases ***
+
+Get signal
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pin}= Get Request gpio_server /api/v1/pin/18 ${headers}
+Log  ${pin.json()}
+Delete All Sessions
+```
+
+For getting all signals you can only delete number of pin on hyperlink.
+Example:
+
+```
+Get signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+That was easier part of our script.
+
+To set values for GPIO's is designed patch method.
+First step is creating dictionary with value to send. It is not difficult:
+```
+${message}= Create Dictionary value 0
+```
+
+`Patch` request requires message data for send.
+
+```
+Set 0 signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 0
+${pin}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pin.json()}
+Delete All Sessions
+```
+
+The same test we can use for setting high values on GPIO. (?)For indicator I used LED
+diodes connected serial with resistors.
+
+Full version of my code
+
+```
+*** Settings ***
+Library RequestsLibrary
+Library Collections
+
+
+*** Variables ***
+${URL} http://192.168.0.100:5000
+
+*** Test Cases ***
+
+Get signal
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin/18 ${headers}
+Log ${pinout.json()}
+
+Set value 1
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 1
+${pinout}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pinout.json()}
+Sleep 3 seconds
+
+Set value 0
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${message}= Create Dictionary value 0
+${pinout}= Patch Request gpio_server /api/v1/pin/18 ${message} headers=${headers}
+Log ${pinout.json()}
+
+Get all signals
+Create Session gpio_server ${URL} verify=True
+${headers}= Create Dictionary Content-Type=application/json Accept=application/json
+${pinout}= Get Request gpio_server /api/v1/pin ${headers}
+Log ${pinout.json()}
+```
+
+This code hasn't got ideal form but it is not subject of this article.
```

Comment:
This code is not in perfect form, but it's not the subject of this article. So polishing is left to the reader.